From c67bf716095765b765bd8dd2be60932ba1966605 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 105/296] check whether the ActivityClientRecord of token is
 null when handleStopActivity.

This is to fix the nullpointer problem happen in the application process because
of activity of the token is not found.

Test: monkey test 10h with no reference problem happen.
---
 core/java/android/app/ActivityThread.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/java/android/app/ActivityThread.java b/core/java/android/app/ActivityThread.java
index 08d7e2a4bfa..f339a84af9b 100644
--- a/core/java/android/app/ActivityThread.java
+++ b/core/java/android/app/ActivityThread.java
@@ -4083,6 +4083,10 @@ public final class ActivityThread {
 
     private void handleStopActivity(IBinder token, boolean show, int configChanges, int seq) {
         ActivityClientRecord r = mActivities.get(token);
+        if (r == null) {
+            Log.w(TAG, "handleStopActivity: no activity for token:" + token);
+            return;
+        }
         if (!checkAndUpdateLifecycleSeq(seq, r, "stopActivity")) {
             return;
         }
-- 
2.11.0

